### ISSUE TITLE
Changes in boots to get WorkflowContextList from updated method in Tink

### ISSUE THREAD

**Issue by gauravgahlot** [tinkerbell/tink#238]
With recent changes in `GetWorkflowContexts` method as per the commit https://github.com/tinkerbell/tink/commit/46881647de83f85ea00b2e51385b9be5e10f470e#diff-a223963715d7c23da698d5963075810eL36 boots does not get the intended list of workflows in the tink server. The worker is not able to PXE boot. Boots needs modification as per https://github.com/tinkerbell/tink/pull/237 to complete. 

## Expected Behaviour
Boots should get the correct list of workflowcontext lists from tink. 

## Current Behaviour
![pxe-fail](https://user-images.githubusercontent.com/28539256/89509320-cd02fc00-d7ec-11ea-97b3-3e1d0a11bf38.png)

Boots logs:
```
{"level":"info","ts":1596648510.5594132,"caller":"src/tftp.go:60","msg":"","service":"github.com/tinkerbell/boots","pkg":"main","client":"192.168.1.5","event":"open","filename":"undionly.kpxe","error":"retrieved job is empty: fetching the workflow","errorVerbose":"fetching the workflow\ngithub.com/tinkerbell/boots/packet.(*Client).GetWorkflowsFromTink\n\t/drone/src/packet/endpoints.go:54\ngithub.com/tinkerbell/boots/job.CreateFromIP\n\t/drone/src/job/job.go:106\nmain.tftpHandler.ReadFile\n\t/drone/src/tftp.go:51\ngithub.com/betawaffle/tftp-go.(*session).serveRRQ\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:228\ngithub.com/betawaffle/tftp-go.(*session).serve\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:158\ngithub.com/betawaffle/tftp-go.serve\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:95\ngithub.com/betawaffle/tftp-go.Serve.func1\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/server.go:164\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357\nretrieved job is empty"}
{"level":"info","ts":1596648510.5598457,"caller":"src/tftp.go:70","msg":"","service":"github.com/tinkerbell/boots","pkg":"main","client":"192.168.1.5","event":"open","filename":"undionly.kpxe","error":"access_violation: permission denied","errorVerbose":"permission denied\naccess_violation\nmain.tftpHandler.ReadFile\n\t/drone/src/tftp.go:70\ngithub.com/betawaffle/tftp-go.(*session).serveRRQ\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:228\ngithub.com/betawaffle/tftp-go.(*session).serve\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:158\ngithub.com/betawaffle/tftp-go.serve\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/handler.go:95\ngithub.com/betawaffle/tftp-go.Serve.func1\n\t/go/pkg/mod/github.com/betawaffle/tftp-go@v0.0.0-20160921192434-dc649c1318ff/server.go:164\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"}
```

## Possible Solution
This could be due to the changes made for the issue #222.
## Steps to Reproduce (for bugs)
1. Setup locally with Vagrant
2. Define a workflow (say Hello-World)
3. Boot the worker machine
4. Observe logs in Boots

## Your Environment

* Operating System and version? : Linux (Ubuntu 18.04)

* How are you running Tinkerbell? : Using Vagrant & VirtualBox

**Comment by Cbkhare** [comment]
Separate method is added as per https://github.com/tinkerbell/tink/pull/237 to get WorkflowContext lists. Boots to be modified accordingly.

**Comment by Cbkhare** [comment]
Tested code changes with #237 `tink` PR. Workflow execution is now working fine. 

```
{"level":"info","ts":1596708550.9102209,"caller":"job/job.go:86","msg":"discovering from ip","service":"github.com/tinkerbell/boots","pkg":"http","ip":"192.168.1.5"}
{"level":"info","ts":1596708550.9137578,"caller":"job/job.go:105","msg":"fetching workflows for hardware","service":"github.com/tinkerbell/boots","pkg":"http","hardwareID":"0eba0bf8-3772-4b4a-ab9f-6ebe93b90a94"}
{"level":"info","ts":1596708550.9242241,"caller":"job/job.go:44","msg":"found active workflow for hardware","service":"github.com/tinkerbell/boots","pkg":"http","workflowID":"49006db0-2737-4b5c-8d44-b8018c9c958a"}
```

**Comment by Cbkhare** [comment]
Riased PR  https://github.com/tinkerbell/boots/pull/64

**Comment by Cbkhare** [comment]
Pr merged tinkerbell/boots#64
